wger/weight/views.py

```diff
@@ -150,9 +150,29 @@ def overview(request, username=None):
                                      'month': max_date.month,
                                      'day': max_date.day}
 
+    last_five_weight_entries = WeightEntry.objects.filter(user=user).order_by('-date')[:5]
+    last_five_weight_entries_details = []
+
+    if last_five_weight_entries:
+        i = 0 
+
+        while (i != len(last_five_weight_entries)-1):
+            curr_entry = last_five_weight_entries[i]
+            prev_entry = last_five_weight_entries[i+1]
+            weight_diff = curr_entry.weight - prev_entry.weight 
+            if weight_diff > 0:
+                weight_diff = '+'+str(weight_diff)
```

Comment:
It's all fixed. I was looking into weight instead of weight_diff for the
"+" sign.


On Sun, Jun 7, 2015 at 1:33 PM, Yu Yu Aung <yuyuaung991@gmail.com> wrote:

> Ok that's a good idea. I'm making changes and pushed by mistake. Please
> ignore the current pull request for now.
>
> Best regards,
> Yu Yu Aung
>
> On Sun, Jun 7, 2015 at 12:34 PM, Roland Geider <notifications@github.com>
> wrote:
>
>> In wger/weight/views.py
>> <https://github.com/rolandgeider/wger/pull/147#discussion_r31878297>:
>>
>> > @@ -150,9 +150,29 @@ def overview(request, username=None):
>> >                                       'month': max_date.month,
>> >                                       'day': max_date.day}
>> >
>> > +    last_five_weight_entries = WeightEntry.objects.filter(user=user).order_by('-date')[:5]
>> > +    last_five_weight_entries_details = []
>> > +
>> > +    if last_five_weight_entries:
>> > +        i = 0
>> > +
>> > +        while (i != len(last_five_weight_entries)-1):
>> > +            curr_entry = last_five_weight_entries[i]
>> > +            prev_entry = last_five_weight_entries[i+1]
>> > +            weight_diff = curr_entry.weight - prev_entry.weight
>> > +            if weight_diff > 0:
>> > +                weight_diff = '+'+str(weight_diff)
>>
>> Actually, we don't need a helper function at all, in the view we can just
>> do weight_diff = day_diff = None. Then in the template we can check if
>> the value is set, something like this
>>
>> {% if entry_detail.1 %}
>>     {% if entry_detail.1 > 0 %}
>>         +
>>     {% endif %}
>>     {{ entry_detail.1 }}
>> {% else %}
>>     -/-
>> {% endif %}
>>
>> This also makes is a bit clearer in the future to know where the
>> "special" values + and -/- come from, since it is all done in the template
>>
>> —
>> Reply to this email directly or view it on GitHub
>> <https://github.com/rolandgeider/wger/pull/147/files#r31878297>.
>>
>
>